generated/provider_dependencies.json

```diff
@@ -26,7 +26,7 @@
   "amazon": {
     "deps": [
       "PyAthena>=3.0.10",
-      "apache-airflow-providers-common-compat>=1.2.1",
+      "apache-airflow-providers-common-compat>=1.3.0",
```

Comment:
We've not yet get 1.3.0 released. Is it ok to just change it here?

Reply:
Yes. This is what chicke-negg provider change enables (this feature was especially implemented for that purpose). In case provider is "chicken-egg" providers list it is installed from local sources instead of from PyPI when generating "PyPI" constraints (this is what has been failing before adding it there).

Since we are releasing the two providers together, they will appear in PyPI at the same time. And in case we decide we do not release common-compat becasue of an error, we will also remove the amazon provider from RC1 and release both from RC2. 

We've been releasing many dependent providers like that in the past.

Reply:
I might add a better description on that in the contributing docs, I think documentation on chicken-egg providers feature is very sparse.

Reply:
Updated docs here https://github.com/apache/airflow/pull/44720